### PR TITLE
Update Assignment 3 & 4 Yichun_Sun

### PR DESCRIPTION
**Pull request by yichun77**
I'm sorry I don't know how to separate different assignment.
Assignment 3: (The 3rd commit) Advanced set of SPARQL queries
Assignment 4: (The 6th & 8th commit) Task 6 and 7 updated

### CODE REVIEW

**Comment by dgarijo** [comment]
I think we are not accepting this one anymore. @ocorcho ?

**Comment by yichun77** [comment]
In Oscar's email to us, he indicated that the deadline for this assignment is October 1st at 23:59, and that the final date in Moodle is also October 1st.

**Comment by ocorcho** [comment]
It's ok for assignment 3.
@dgarijo if you find the assignment 4 ok, then you can accept the PR.

File: Assignment4/yichun77-22A350/Task06.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1uLM1eXxDo2jWT5P-JkGkLk51t0GcjjPC
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/yichun77-22A350/Task06.py:12]: This won't work in Python. It's notebook magic

File: Assignment4/yichun77-22A350/Task07.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+for s_sub,p_sub,o_sub in g.triples((None, RDFS.subClassOf, ns.Person)):
+  for s_subIndividual,ps_subIndividual,os_subIndividual in g.triples((None, RDF.type, s_sub)):
+    print("RDFLib - Individuals of subclasses of Person:", s_subIndividual)
+
+# SPARQL
+q_person = prepareQuery('''
+  SELECT ?s WHERE { 
+    {?s rdf:type ns:Person} 
+    UNION{
+      ?s rdfs:subClassOf ns:Person .
```
Comment (dgarijo) [inline Assignment4/yichun77-22A350/Task07.py:59]: Take all the subclasses of person and its subclasses. I.e., extract all the taxonomy with *

File: Assignment4/yichun77-22A350/Task07.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      print("RDFLib - Individuals and properties of subclasses of Person:", s_pro,p_pro,o_pro)
+
+# SPARQL
+q_pro = prepareQuery('''
+  SELECT ?s ?p ?o WHERE { 
+    {
+      ?s rdf:type ns:Person .
+      ?s ?p ?o
+    } UNION {
+      ?sub rdfs:subClassOf ns:Person .
```
Comment (dgarijo) [inline Assignment4/yichun77-22A350/Task07.py:88]: same as above

**Review by dgarijo — CHANGES_REQUESTED** [review]

File: Assignment4/yichun77-22A350/Task07.py (hunk 3 above)
Comment (yichun77) [reply Assignment4/yichun77-22A350/Task07.py]: Many thanks for the comments. All corrections are done.

**Review by dgarijo — APPROVED** [review]

**Pull request merged**